tests/raftstore/pd.rs

```diff
@@ -151,45 +141,215 @@ impl Cluster {
     fn get_stores(&self) -> Vec<metapb::Store> {
         self.stores.values().map(|s| s.store.clone()).collect()
     }
+
+    fn handle_heartbeat_version(&mut self, region: metapb::Region) -> Result<()> {
+        // For split, we should handle heartbeat carefully.
+        // E.g, for region 1 [a, c) -> 1 [a, b) + 2 [b, c).
+        // after split, region 1 and 2 will do heartbeat independently.
+        let end_key = enc_end_key(&region);
+        let start_key = enc_start_key(&region);
+        assert!(end_key > start_key);
+
+        let version = region.get_region_epoch().get_version();
+        let conf_ver = region.get_region_epoch().get_conf_ver();
+
+        let search_key = data_key(region.get_start_key());
+        let search_region = self.get_region(search_key).unwrap();
+        let search_end_key = enc_end_key(&search_region);
+        let search_start_key = enc_start_key(&search_region);
+        let search_version = search_region.get_region_epoch().get_version();
+        let search_conf_ver = search_region.get_region_epoch().get_conf_ver();
+
+        if start_key == search_start_key && end_key == search_end_key {
+            // we are the same, must check epoch here.
+            try!(check_stale_region(&search_region, &region));
+        } else if search_start_key > end_key {
```

Comment:
Btw, I find no need to check following empty end key here, because `search_start_key > end_key` here can check it already.